doc/conf.py

```diff
@@ -96,6 +103,15 @@
 
 autodoc_docstring_signature = True
 
+
+# Sphinx gallery configuration
+sphinx_gallery_conf = {
+   # path to your examples scripts
+    'examples_dirs' : '../examples',
+    # path where to save gallery generated examples
+    'gallery_dirs'  : 'auto_examples'}
```

Comment:
Can we not put it in `gallery` instead? Or if it were `mpl_examples`, then some of the lines that were changed would not have been necessary.

Whatever it becomes, I don't like the `auto_` prefix; it's an internal concern and doesn't really make any sense to present that in the URL to a user from a semantic point of view.